cheevos-new/cheevos.c

```diff
@@ -680,20 +685,41 @@ static void rcheevos_test_cheevo_set(bool official)
       count = rcheevos_locals.patchdata.unofficial_count;
    }
 
+   rcheevos_locals.invalid_peek_address = false;
+
    for (i = 0; i < count; i++, cheevo++)
    {
       /* Check if the achievement is active for the current mode. */
-      if ((cheevo->active & mode) == 0)
-         continue;
-
       if (cheevo->active & mode)
       {
          int valid = rc_test_trigger(cheevo->trigger, rcheevos_peek, NULL, NULL);
 
+         /* trigger must be false for at least one frame before it can trigger. if last is true, the trigger hasn't yet been false. */
          if (cheevo->last)
+         {
+            /* if the we're still waiting for the trigger to stabilize, check to see if an error occurred */
+            if (rcheevos_locals.invalid_peek_address)
+            {
+               /* could not map one or more addresses - disable the achievement */
+               CHEEVOS_ERR(RCHEEVOS_TAG "Achievement disabled (invalid address): %s\n", cheevo->info->title);
+               cheevo->active = 0;
+
+               /* clear out the trigger so it shows up as 'Unsupported' in the menu */
+               CHEEVOS_FREE(cheevo->trigger);
+               cheevo->trigger = NULL;
+
+               /* reset the flag for the next achievement */
+               rcheevos_locals.invalid_peek_address = false;
+               continue;
+            }
+
+            /* no error, reset any hit counts for the next check */
             rc_reset_trigger(cheevo->trigger);
+         }
          else if (valid)
+         {
```

Comment:
In the future, if a code block just consists of one line and we aren't calling a macro on that line, just do away with the brackets.

Reply:
I usually only use brackets around single statements to pair with brackets around a group of statements in the other part of an if/else chain. I will remove these as recommended.